Mathlib/MeasureTheory/Measure/EverywherePos.lean

```diff
@@ -0,0 +1,296 @@
+* `isEverywherePos_everywherePosSubset` shows that `μ.everywherePosSubset s` satisfies the property
+  `μ.IsEverywherePos` if `μ` is inner regular and `s` is measurable.
+
+The latter two statements have also versions when `μ` is inner regular for finite measure sets,
+assuming additionally that `s` has finite measure.
+
+* `IsEverywherePos.IsGδ` proves that an everywhere positive compact closed set is a Gδ set,
+  in a topological group with a left-invariant measure. This is a nontrivial statement, used
+  crucially in the study of the uniqueness of Haar measures.
+* `innerRegularWRT_preimage_one_hasCompactSupport_measure_ne_top`: for a Haar measure, any
+  finite measure set can be approximated from inside by level sets of continuous
+  compactly supported functions. This property is also known as completion-regularity of Haar
+  measures.
+-/
+
+open scoped Topology ENNReal NNReal
+open Set Filter
+
+namespace MeasureTheory.Measure
+
+variable {α : Type*} [TopologicalSpace α] [MeasurableSpace α]
+
+/-- A set `s` is *everywhere positive* (also called *self-supporting*) with respect to a
+measure `μ` if it has positive measure around each of its points, i.e., if all neighborhoods `n`
+of points of `s` satisfy `μ (s ∩ n) > 0`. -/
+@[pp_dot] def IsEverywherePos (μ : Measure α) (s : Set α) : Prop :=
+  ∀ x ∈ s, ∀ n ∈ 𝓝[s] x, 0 < μ n
+
+/-- * The everywhere positive subset of a set is the subset made of those points all of whose
+neighborhoods have positive measure inside the set. -/
+@[pp_dot] def everywherePosSubset (μ : Measure α) (s : Set α) : Set α :=
+  {x | x ∈ s ∧ ∀ n ∈ 𝓝[s] x, 0 < μ n}
+
+lemma everywherePosSubset_subset (μ : Measure α) (s : Set α) : μ.everywherePosSubset s ⊆ s :=
+  fun _x hx ↦ hx.1
+
+/-- The everywhere positive subset of a set is obtained by removing an open set. -/
+lemma exists_isOpen_everywherePosSubset_eq_diff (μ : Measure α) (s : Set α) :
+    ∃ u, IsOpen u ∧ μ.everywherePosSubset s = s \ u := by
+  refine ⟨{x | ∃ n ∈ 𝓝[s] x, μ n = 0}, ?_, by ext x; simp [everywherePosSubset, zero_lt_iff]⟩
+  rw [isOpen_iff_mem_nhds]
+  intro x ⟨n, ns, hx⟩
+  rcases mem_nhdsWithin_iff_exists_mem_nhds_inter.1 ns with ⟨v, vx, hv⟩
+  rcases mem_nhds_iff.1 vx with ⟨w, wv, w_open, xw⟩
+  have A : w ⊆ {x | ∃ n ∈ 𝓝[s] x, μ n = 0} := by
+    intro y yw
+    refine ⟨s ∩ w, inter_mem_nhdsWithin _ (w_open.mem_nhds yw), measure_mono_null ?_ hx⟩
+    rw [inter_comm]
+    exact (inter_subset_inter_left _ wv).trans hv
+  have B : w ∈ 𝓝 x := w_open.mem_nhds xw
+  exact mem_of_superset B A
+
+variable {μ ν : Measure α} {s k : Set α}
+
+protected lemma _root_.MeasurableSet.everywherePosSubset [OpensMeasurableSpace α]
+    (hs : MeasurableSet s) :
+    MeasurableSet (μ.everywherePosSubset s) := by
+  rcases exists_isOpen_everywherePosSubset_eq_diff μ s with ⟨u, u_open, hu⟩
+  rw [hu]
+  exact hs.diff u_open.measurableSet
+
+protected lemma _root_.IsClosed.everywherePosSubset (hs : IsClosed s) :
```

Comment:
The support is closed, that's a general property, so the support will be closed in `s`, for the induced topology.

Reply:
Yes, that's what the previous lemma says.